primer/brand
# VideoPlayer component improvements

## Summary

A refactor of the `VideoPlayer` component to make it more modular and customisable. The refactor introduces new hooks and controls to manage video state and provide a more flexible API for customising the video player.

## List of notable changes:

- allow video title to be hidden via the `visuallyHiddenTitle` prop
- allow a custom play overlay to be provided via the `renderPlayOverlay` prop
- refactored video controls into standalone components. Split some of the controls CSS into separate files, but styling would benefit from a more comprehensive refactor in the future
- allow video controls to be customised via the `renderControls` prop
- added new `useVideo` hook and context to manage video state
- added new `useIsElementFullScreen` hook to manage fullscreen state
- added new `useVideoKeypressHandler` hook to add keyboard shortcuts to control video (see updated documentation for specific shortcuts)
- added new `useVideoResizeObserver` hook to consolidate resize logic. When Safari 18 drops later this year this hook can be removed in favour of container queries
- updated custom `useRef` hook to accept `ForwardedRef` types as well
- updated Storybook with more examples
- updated documentation
- updated test suite

## What should reviewers focus on?

- Look and feel of the video player
- Test it and see if you run into any issues
- Check the documentation and Storybook for the new examples
- Make sure the API changes are clear and easy to understand

## Steps to test:

[View the Storybook here](https://primer-2c317f85db-26139705.drafts.github.io/storybook/?path=/story/components-videoplayer-features--with-poster)


## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">



 </td>
<td valign="top">



</td>
</tr>
</table>
